source/common/network/addr_family_aware_socket_option_impl.cc

```diff
@@ -11,17 +11,24 @@ namespace Envoy {
 namespace Network {
 
 namespace {
+Address::IpVersion getVersionFromAddress(Address::InstanceConstSharedPtr addr) {
+  if (addr->ip() != nullptr) {
+    return addr->ip()->version();
+  }
+  throw EnvoyException("Unable to set socket option on non-IP sockets");
```

Comment:
Should this method throw a more generic `EnvoyException`, followed by `setOption` throwing this one?

Reply:
This is caught in `getVersionFromSocket()` and converted to a nullopt return.

Reply:
I just felt that the exception message didn't match the function's operation.

Reply:
Yeah, technically it doesn't, but I don't think it's a big deal given the hyper local use here.